### PR TITLE
test : getStringFromServiceAnnotation() 

### PR DESCRIPTION
**Pull request by KingDaemonX**
**What this PR does / why we need it**:

**Which issue this PR fixes(if applicable)**:
fixes #

**Special notes for reviewers**:

**Release note**:

```release-note
NONE
```

cover all possible edge cases that can during the use of this function and test for behavour of function giving hardcoded value incase there is a change in the logic of the function

### CODE REVIEW

File: pkg/openstack/loadbalancer_test.go [hunk 1]
```diff
@@ -534,3 +535,153 @@ func Test_getListenerProtocol(t *testing.T) {
...
+			},
+			expect: "annotation-Value",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if got := getStringFromServiceAnnotation(test.testArgs.service, test.testArgs.annotationKey, test.testArgs.defaultSetting); got != test.expect {
+				t.Errorf("[Error] : getStringFromServiceAnnotation() => %s, expecting = %s", got, test.expect)
+			}
```
Comment (dulek) [inline pkg/openstack/loadbalancer_test.go:684]: This should be done through `assert.Equal()`.

Comment (KingDaemonX) [reply pkg/openstack/loadbalancer_test.go]: noted! and fixed

File: pkg/openstack/loadbalancer_test.go [hunk 2]
```diff
@@ -534,3 +535,153 @@ func Test_getListenerProtocol(t *testing.T) {
...
+						Namespace:   "service-namespace",
+						Name:        "service-name",
+						Annotations: map[string]string{"annotationKey": "annotation-Value"},
+					},
+				},
+				annotationKey:  "annotationKey",
+				defaultSetting: "default-setting",
+			},
+			expected: "annotation-Value",
+		},
```
Comment (dulek) [inline pkg/openstack/loadbalancer_test.go:605]: This seems to be a duplicate of the case number 2 - with the same name.

**Review by dulek — CHANGES_REQUESTED** [review]

**Review by pierreprinetti — APPROVED** [review]
LGTM! Please rebase and we're good to merge.

**Comment by pierreprinetti** [comment]
My bad, this is rebased already. Let's go!

/lgtm
